framework/src/ble_manager/ble_manager_api.c

```diff
@@ -0,0 +1,343 @@
+/****************************************************************************
+ *
+ * Copyright 2021 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+#include <tinyara/config.h>
+
+#include <stdio.h>
+#include <stdint.h>
+#include <ble_manager/ble_manager.h>
+#include "ble_manager_event.h"
+#include "ble_manager_msghandler.h"
+
+#define RETURN_RESULT(res, msg)      \
+	do {                             \
+		if (res < 0) {               \
+			return BLE_MANAGER_FAIL; \
+		} else {                     \
+			return msg.result;       \
+		}                            \
+	} while (0)
+
+ble_result ble_manager_init(ble_client_init_config *client_config, ble_server_init_config *server_config)
+{
+	if (client_config == NULL && server_config == NULL) {
```

Comment:
The only error case is that both of the params are NULL.
So I checked them with "&&" operator.

Reply:
I don't think so, you need to divide to ble_manager_client_init & ble_manager_server_init, because both of config is already there, also config includes pointers of each function.
Otherwise, you'd better make common structure include variable which represents type(server or client)

Reply:
Actually, an user cannot choose the mode of BLE, because it is fixed in build time.
So I will check it in HAL layer instead of BLE framework.